gensim/models/fasttext.py

```diff
@@ -918,153 +871,46 @@ def load(cls, *args, **kwargs):
             Save :class:`~gensim.models.fasttext.FastText` model.
 
         """
-        try:
-            model = super(FastText, cls).load(*args, **kwargs)
-
-            if not hasattr(model.trainables, 'vectors_vocab_lockf') and hasattr(model.wv, 'vectors_vocab'):
-                model.trainables.vectors_vocab_lockf = ones(model.wv.vectors_vocab.shape, dtype=REAL)
-            if not hasattr(model.trainables, 'vectors_ngrams_lockf') and hasattr(model.wv, 'vectors_ngrams'):
-                model.trainables.vectors_ngrams_lockf = ones(model.wv.vectors_ngrams.shape, dtype=REAL)
-
-            if not hasattr(model.wv, 'bucket'):
-                model.wv.bucket = model.trainables.bucket
-        except AttributeError:
-            logger.info('Model saved using code from earlier Gensim Version. Re-loading old model in a compatible way.')
-            from gensim.models.deprecated.fasttext import load_old_fasttext
-            model = load_old_fasttext(*args, **kwargs)
-
-        gensim.models.keyedvectors._try_upgrade(model.wv)
+        model = super(FastText, cls).load(*args, rethrow=True, **kwargs)
+
+        if not hasattr(model.wv, 'vectors_vocab_lockf') and hasattr(model.wv, 'vectors_vocab'):
+            # TODO: try trainables-location
+            model.wv.vectors_vocab_lockf = ones(1, dtype=REAL)
+        if not hasattr(model, 'vectors_ngrams_lockf') and hasattr(model.wv, 'vectors_ngrams'):
+            # TODO: try trainables-location
+            model.wv.vectors_ngrams_lockf = ones(1, dtype=REAL)
+        # fixup mistakenly overdimensioned gensim-3.x lockf arrays
+        if len(model.wv.vectors_vocab_lockf.shape) > 1:
+            model.wv.vectors_vocab_lockf = ones(1, dtype=REAL)
```

Comment:
Yes, I think so too. The gensim FastText implementation fumbled the porting of this W2V/D2V experiment (overallocating its lockf arrays by a factor of `vector_size`, which has massively bloated gensim FT models), so anyone genuinely trying to use this feature might have failed, or at least noticed & reported this overallocation. I thus regret ever adding it, and seriously considered removing it. But now that it's fixed to be of just a few bytes' cost (in FT & W2V/D2V) if unused, and I've added more EXPERIMENTAL warnings in comments/docs, and given that it might be of use to the constant requests for 'fine-tuning' functionality, and that there's a chance someone's used it, I decided not to actually remove `lockf` entirely.